python/ecl/summary/ecl_sum.py

```diff
@@ -524,6 +525,59 @@ def pandas_frame(self, time_index = None, column_keys = None):
         frame = pandas.DataFrame(index = time_index, columns=list(keywords), data=data)
         return frame
 
+    @staticmethod
+    def _compile_headers_list(headers, dims):
+        var_list = []
+        for key in headers:
+            lst = re.split(':', key)
+            kw = lst[0]
+            wgname = None
+            num = 0;
+            unit = "UNIT"
+            if len(lst) > 1:
+                nums = []
+                if lst[1][0].isdigit():       
+                    nums = re.split(',', lst[1])
+                else:
+                    wgname = lst[1]
+                if len(lst) == 3:
+                    nums = re.split(",", lst[2])
+                if len(nums) == 3:
+                    i = int(nums[0])-1
+                    j = int(nums[1])-1
+                    k = int(nums[2])-1
+                    if dims is None:
+                        raise ValueError("For key %s When using indexing i,j,k you must supply a valid value for the dims argument" % key)
+                    num = i + j * dims[0] + k * dims[0]*dims[1] + 1
+                elif len(nums) == 1:
+                    num = int(nums[0])
+                      
+            var_list.append( [kw, wgname, num, unit] )     
+        return var_list   
+
+    @classmethod
+    def from_pandas(cls, case, frame, dims = None, headers = None):
+        start_time = frame.index[0]
+        ecl_sum = EclSum.writer(case,
```

Comment:
This will fail if `dims is None`. I suggest something like this:
```
if dims is None:
    nx = 1; ny = 1; nz = 1
else:
    nx = dims[0]; ny = dims[1]; nz = dims[2]

ecl_sum = EclSum.writer(case, start_time.to_pydatetime(), nx, ny, nz)
```